### PR TITLE
1.13 support, general refactor

### PR DESCRIPTION
**Pull request by drewdorris**
Added 1.13 support and refactored much of the project. Little of the core code was changed; instead, mainly appearance & usability was adjusted for easier usage of the API
* Add "Sound" enum with all current and previous note enum values to more readily handle changes in org.bukkit.Sound
* Add javadocs - I don't know the full extent of this API so I left some things alone
* Deprecate "Instrument", rename to "InstrumentUtils" to avoid confusing the class as an instantiable object
* Rename bunches of variables and methods for less confusion
* Use tabs instead of spaces & use consistent curly brace usage
* Use UUIDs instead of deprecated player name usage
* Use pom for setting version/main class/etc in plugin.yml

### CODE REVIEW

File: src/main/java/com/xxmicloxx/NoteBlockAPI/Instrument.java [hunk 1]
```diff
@@ -1,97 +1,114 @@
...
+				return org.bukkit.Instrument.BASS_GUITAR;
+			case 2:
+				return org.bukkit.Instrument.BASS_DRUM;
+			case 3:
+				return org.bukkit.Instrument.SNARE_DRUM;
+			case 4:
+				return org.bukkit.Instrument.STICKS;
+			case 5:
+				return org.bukkit.Instrument.GUITAR;
+			case 6:
```
Comment (koca2000) [inline src/main/java/com/xxmicloxx/NoteBlockAPI/Instrument.java]: Instruments below are not in Miinecraft 1.8

File: src/main/java/com/xxmicloxx/NoteBlockAPI/InstrumentUtils.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				return org.bukkit.Instrument.BASS_GUITAR;
+			case 2:
+				return org.bukkit.Instrument.BASS_DRUM;
+			case 3:
+				return org.bukkit.Instrument.SNARE_DRUM;
+			case 4:
+				return org.bukkit.Instrument.STICKS;
+			case 5:
+				return org.bukkit.Instrument.GUITAR;
+			case 6:
```
Comment (koca2000) [inline src/main/java/com/xxmicloxx/NoteBlockAPI/InstrumentUtils.java]: Same as in Instrument

**Comment by drewdorris** [comment]
Missed that, fixed

**Comment by koca2000** [comment]
I have just take a deeper look. Why are you using `getInstrumentName` in `InstrumentUtils`? It will cost much more time to proceed.

**Comment by drewdorris** [comment]
I was experiencing issues with using NoteBlockAPI with one version and another plugin that relies on the API with a different version -- the plugin would use `getInstrument` to get the Sound enum for an instrument, but the enum would not exist in the shaded Bukkit API for the other plugin, instead being a version ahead. I added the `getInstrumentName` method so that it would be possible to receive the name of a Sound enum without actually retrieving a potentially non-existent Sound enum. With that, I'm able to parse the name and retrieve the Sound enum for the other plugin. I found the method was needed for myself, and I imagine others may need it as well in certain situations. There shouldn't be any noticeable performance difference using `getInstrumentName` in `getInstrument`, as it only enters a short loop to retrieve the value, which should be virtually instant.

I used `getFromBukkitName` in `getInstrumentName` because the name of the enum in the local Sound class is not necessarily the latest updated name of the org.bukkit.Sound enum name, it's just one of the ones in the past. `getFromBukkitName` retrieves the one relevant for the latest version

**Pull request merged**